test/scaling.jl

```diff
@@ -0,0 +1,393 @@
+
+                @test collect(transformed) ≈ collect(nt_expected) atol=1e-5
+                @test inverted ≈ [nt.a, nt.b] atol=1e-5
+            end
+        end
+
+        @testset "DataFrame" begin
+            df = DataFrame(:a => [0.0, -0.5, 0.5], :b => [1.0, 0.0, 2.0])
+            df_expected = DataFrame(:a => [0.0, -1.0, 1.0], :b => [0.0, -1.0, 1.0])
+
+            @testset "Non-mutating" begin
+                scaling = MeanStdScaling(df)
+                transformed = Transforms.apply(df, scaling)
+                @test transformed ≈ [df_expected.a, df_expected.b] atol=1e-5
+                @test scaling(df) ≈ [df_expected.a, df_expected.b] atol=1e-5
+            end
+
+            @testset "Mutating" begin
+                scaling = MeanStdScaling(df)
+                _df = deepcopy(df)
+                Transforms.apply!(_df, scaling)
+                @test _df isa DataFrame
+                @test _df ≈ df_expected atol=1e-5
+            end
+
+            @testset "cols = $c" for c in (:a, :b)
+                scaling = MeanStdScaling(df; cols=[c])
+
+                transformed = Transforms.apply(df, scaling; cols=[c])
+                @test transformed ≈ [df_expected[!, c]] atol=1e-5
+
+                _df = deepcopy(df)
+                _df_expected = deepcopy(df)
+                _df_expected[!, c] = df_expected[!, c]
+                Transforms.apply!(_df, scaling; cols=[c])
+                @test _df isa DataFrame
+                @test _df ≈ _df_expected atol=1e-5
+            end
+
+            @testset "Re-apply" begin
+                scaling = MeanStdScaling(df)
+                Transforms.apply(df, scaling)
+
+                # Expect scaling parameters to be fixed to the first data applied to
+                df2 = DataFrame(:a => [-1.0, 0.5, 0.0], :b => [2.0, 2.0, 1.0])
+                df_expected2 = DataFrame(:a => [-2.0, 1.0, 0.0], :b => [1.0, 1.0, 0.0])
+                @test ≈(
+                    Transforms.apply(df2, scaling),
+                    [df_expected2.a, df_expected2.b],
+                    atol=1e-5
+                )
+            end
+
+            @testset "Inverse" begin
+                scaling = MeanStdScaling(df)
+                transformed = Transforms.apply(df, scaling)
+                transformed = DataFrame(:a => transformed[1], :b => transformed[2])
+                inverted = Transforms.apply(transformed, scaling; inverse=true)
+
+                @test transformed ≈ df_expected atol=1e-5
+                @test inverted ≈ [df.a, df.b] atol=1e-5
+            end
+        end
+
+        @testset "Zero std" begin
+            x = ones(Float64, 3)
+            expected = zeros(Float64, 3)
+
+            scaling = MeanStdScaling(x)
+
+            @test Transforms.apply(x, scaling) == expected  # default `eps`
+            @test Transforms.apply(x, scaling; eps=1) == expected
+            @test all(isnan.(Transforms.apply(x, scaling; eps=0)))  # 0/0
+        end
+    end
+end
```

Comment:
Does this transform support inds? Would be nice to get a few tests for that as well if it does.

Reply:
Not currently, but it can support inds. I'm not sure what it would be used for though. If we can get it working with the general `apply` method, then inds should be supported.